### ISSUE TITLE
Externalization of ag-grid package using webpack + angular - Worked with small patch in ag-grid

### ISSUE THREAD

**Issue by hardikpatel043** [ag-grid/ag-grid#4074]
**I'm submitting a ...** 
```
[] bug report => see 'Providing a Reproducible Scenario'
[] feature request => do not use Github for feature requests, see 'Customers of ag-Grid'
[X] support request => see 'Requesting Community Support'
```
I am trying to use ag-grid in my micro front end angular app. I have multiple micro apps and I use ngx-build-plus to build each micro app. To share all third party dependencies(for e.g. ag-grid) between micro apps, I am using webpack externals config.

This means I am not bundling ag-grid with each micro front end. Shell app will have one copy of ag-grid scripts and all other micro apps uses same. This is my goal.

To do this I have specified webpack external config for my micro app as below.

```
const webpack = require('webpack');
module.exports = {
  externals: {
    rxjs: 'rxjs',
    '@angular/core': 'ng.core',
    '@angular/common': 'ng.common',
    '@angular/common/http': 'ng.common.http',
    '@angular/platform-browser': 'ng.platformBrowser',
    '@angular/platform-browser-dynamic': 'ng.platformBrowserDynamic',
    '@angular/compiler': 'ng.compiler',
    '@angular/elements': 'ng.elements',
    'ag-grid-community': 'agGridCommunity',
    'ag-grid-angular': 'ag-grid-angular',
    '@angular/router': 'ng.router',
  },
};
```

For my shell app, I mentioned ag-grid script explicitly in angular.json

```
 "scripts": [
              "node_modules/ag-grid-community/dist/ag-grid-community.js",
              "node_modules/ag-grid-angular/bundles/ag-grid-angular.umd.js"
            ]
```

**Current behavior**
When I load my micro app, it gives me below error.

![image](https://user-images.githubusercontent.com/13534287/93577463-b63df080-f9b9-11ea-8c4c-e22048bece79.png)

**Expected behavior**
I should be able to see grid.

**My Observation:** 

My micro app bundle looks like below
![image](https://user-images.githubusercontent.com/13534287/93577626-e5546200-f9b9-11ea-8ea5-38c02f088790.png)

The reason of this error is the name of the package.

So, I tried to rename the package name from ag-grid-angular to ag.grid.angular in ag-grid-angular.umd.js file as below

before:
![image](https://user-images.githubusercontent.com/13534287/93578078-7aeff180-f9ba-11ea-9385-5cef4ad8a62c.png)

after:
![image](https://user-images.githubusercontent.com/13534287/93578027-6ca1d580-f9ba-11ea-9602-0ce7c3c2c738.png)

This patch worked for me to externalized the ag-grid package. Now my webpack external config look like this.

 ```
const webpack = require('webpack');
module.exports = {
  externals: {
    rxjs: 'rxjs',
    '@angular/core': 'ng.core',
    '@angular/common': 'ng.common',
    '@angular/common/http': 'ng.common.http',
    '@angular/platform-browser': 'ng.platformBrowser',
    '@angular/platform-browser-dynamic': 'ng.platformBrowserDynamic',
    '@angular/compiler': 'ng.compiler',
    '@angular/elements': 'ng.elements',
    'ag-grid-community': 'agGridCommunity',
    'ag-grid-angular': 'ag.grid.angular',
    '@angular/router': 'ng.router',
  },
};
```

Micro app bundle looks like below after patch.

![image](https://user-images.githubusercontent.com/13534287/93577982-58f66f00-f9ba-11ea-9a3c-d57f0e76bd2c.png)

I am using below versions:
```
 "dependencies": {
    "@angular-extensions/elements": "^10.0.2",
    "@angular/animations": "^10.0.0",
    "@angular/cdk": "^10.1.3",
    "@angular/common": "^10.0.0",
    "@angular/compiler": "^10.0.0",
    "@angular/core": "^10.0.0",
    "@angular/elements": "^10.0.14",
    "@angular/flex-layout": "^10.0.0-beta.32",
    "@angular/forms": "^10.0.0",
    "@angular/material": "^10.1.3",
    "@angular/platform-browser": "^10.0.0",
    "@angular/platform-browser-dynamic": "^10.0.0",
    "@angular/router": "^10.0.0",
    "@nrwl/angular": "10.0.12",
    "@nrwl/nx-cloud": "^10.0.0",
    "@webcomponents/custom-elements": "^1.4.2",
    "@webcomponents/webcomponentsjs": "^2.4.4",
    "ag-grid-angular": "^24.0.0",
    "ag-grid-community": "^24.0.0",
    "ngx-build-plus": "^10.1.1",
    "rxjs": "~6.5.5",
    "zone.js": "^0.10.2"
  },
  "devDependencies": {
    "@angular/cli": "~10.0.0",
    "@nrwl/workspace": "10.0.12",

```

Here is my reproducible [code](https://github.com/hardikpatel043/mf-app/tree/fix/ag-grid).

Steps to follow:
1. Clone Repo
2. npm install
3. run command  "npm run build:home:externals". This will create bundle of home micro app.
4. copy main-es2015.js file to cz-app/src/home
5. cz-app is my shell app.
6. run shell app using "ng serve"
7. This will not work. So patch is required as mentioned above in file node_modules\ag-grid-angular\bundles\ag-grid-angular.umd.js.
8. once patch is done, shell app will work fine and able to load grid.

**Conclusion:**

I am not sure how this can be taken with ag-grid package. I wanted to share what I have observed. Someone else might have different way of solving this issue. Need your suggestions.

**Comment by makinggoodsoftware** [comment]
Hi,

Your ticket has been flagged as in_zendesk, it means that we recognize this as an issue worth investigating so we have moved it to our official support channel for customers (zendesk)

We will carry on there with the investigation and we will update you as soon as possible.

Thanks

**Comment by makinggoodsoftware** [comment]
Hi,

I am afraid that trying to follow this:

Clone Repo
npm install
run command "npm run build:home:externals". This will create bundle of home micro app.
copy main-es2015.js file to cz-app/src/home
cz-app is my shell app.
run shell app using "ng serve"
This will not work. So patch is required as mentioned above in file node_modules\ag-grid-angular\bundles\ag-grid-angular.umd.js.

Works for us...

Unfortunately, we will have to close the ticket, if you can consistently reproduce this in your end, feel free to create a new ticket

Thanks

**Comment by hardikpatel043** [comment]
@makinggoodsoftware So, above steps will clone master branch. You need to switch to branch 'fix/ag-grid'.

I have provided a link for this branch in the issue.

Could you please check with this?

**Comment by hardikpatel043** [comment]
@makinggoodsoftware  Can we reopen this issue?

**Comment by hardikpatel043** [comment]
any update of this issue?
